### PR TITLE
MQE-2242: Add static check to flag use of <pause>

### PR DESCRIPTION
**Pull request by soumyau**
### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<issue_number>: Issue title
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: dev/tests/util/MftfStaticTestCase.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            $input->method('getOption')
+                ->with('path')
+                ->willReturn($path);
+        }
+        return $input;
+    }
+
+    public function tearDown(): void
+    {
+        DirSetupUtil::rmdirRecursive(self::STATIC_RESULTS_DIR);
```
Comment (jilu1) [inline dev/tests/util/MftfStaticTestCase.php:55]: Do we need rmDir in both before and tearDown?

File: dev/tests/verification/PauseCheckModule/Test/TestWithMultiplePauseActionsTest.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            <severity value="CRITICAL"/>
+            <group value="functional"/>
+            <features value="Pause check"/>
+            <stories value="MQE-433"/>
+        </annotations>
+        <amOnPage stepKey="step1" url="/step1"/>
+        <fillField stepKey="step2" selector="#username" userInput="step2"/>
+        <pause stepKey="pauseAfterStep2"/>
+        <fillField stepKey="step3" selector="#password" userInput="step3"/>
+        <pause stepKey="pauseAfterStep3"/>
```
Comment (jilu1) [inline dev/tests/verification/PauseCheckModule/Test/TestWithMultiplePauseActionsTest.xml:22]: Possibly add `pause` in <before> and <after> also

File: src/Magento/FunctionalTestingFramework/Util/Script/ScriptUtil.php [hunk 3]
```diff
@@ -101,7 +101,7 @@ public function getModuleXmlFilesByScope($modulePaths, $scope)
             if (!realpath($modulePath . $scopePath)) {
                 continue;
             }
-            $finder->files()->followLinks()->in($modulePath . $scopePath)->name("*.xml");
+            $finder->files()->followLinks()->in($modulePath . $scopePath)->name("*.xml")->sortByName();
```
Comment (jilu1) [inline src/Magento/FunctionalTestingFramework/Util/Script/ScriptUtil.php:104]: Have you checked if this sorting has any side effect on other scripts?

File: dev/tests/util/MftfStaticTestCase.php (hunk 1 above)
Comment (soumyau) [reply dev/tests/util/MftfStaticTestCase.php:47]: agreed, removed setup()

File: src/Magento/FunctionalTestingFramework/Util/Script/ScriptUtil.php (hunk 3 above)
Comment (soumyau) [reply src/Magento/FunctionalTestingFramework/Util/Script/ScriptUtil.php:104]: no side effects observed, it's only used in deprecated entity test which is covered by verification test

**Pull request merged**